src/loader.js

```diff
@@ -24,7 +24,7 @@ module.exports = function(source) {
   const resourcesFromConfig =
     webpack.version !== 2
     ? webpack.options.sassResources
-    : loaderUtils.parseQuery(this.query).resources
+    : (loaderUtils.getOptions(this) || { resources: undefined }).resources
```

Comment:
We don't really need `resources: undefined` here, just use an empty object instead. And let's throw right after this if resources not found.

```js
const isWebpack2 = webpack.version === 2;
const resourcesFromConfig =
  isWebpack2
  ? (loaderUtils.getOptions(this) || {}).resources
  : webpack.options.sassResources;

if (!resourcesFromConfig) {
  const error = new Error(`
    Can't find sass resources in your config.
    Make sure ${isWebpack2 ? 'loader.options.resources' : 'webpackConfig.sassResources'} exists.
  `);

  return callback(error);
}